cpp/modules/luma.gl/garrow/src/field.h

```diff
@@ -21,31 +21,50 @@
 #ifndef LUMAGL_GARROW_FIELD_H
 #define LUMAGL_GARROW_FIELD_H
 
+#include <dawn/webgpu_cpp.h>
+
 #include <memory>
 #include <string>
 #include <utility>
 
-#include "./util/attribute-descriptor.h"
+#include "./key-value-metadata.h"
 
 namespace lumagl {
 namespace garrow {
 
 /// \brief A combination of a field name and additional metadata specific to this field.
 class Field {
  public:
-  // TODO(ilija@unfolded.ai): In order for this to match arrow API, and decouple it from attributes,
-  // we should use general-purpoes metadata instead
-  Field(const std::string& name, const AttributeDescriptor& descriptor)
-      : _name{std::move(name)}, _descriptor{std::move(descriptor)} {}
+  // NOTE: type is currently a simple wgpu::VertexFormat value. Arrow has a complex DataType implementation that
+  // deals with buffer type complexity, which is something that should be put in place once we implement reading
+  Field(const std::string& name, wgpu::VertexFormat type, bool nullable = false,
```

Comment:
Again, I wonder if we should use Arrow types here, instead of VertexFormat type.

I suppose it will depend on how these tables are used. 

But, I am sure you have gone back and forth on this.